README.md

```diff
@@ -1 +1,145 @@
-# angular-http-readme
\ No newline at end of file
+# $http
+
+## Overview
+
+One common requirement of apps is the ability to make HTTP requests. Luckily, Angular provides us a service named `$http` to do these with ease - let's take a look!
+
+## Objectives
+
+- Describe $http
+- Use $http methods
+- Fetch data from an API
+- Update data from an API
+
+## What is $http?
+
+$http is a core Angular service that provides a simplistic API to allow us to communicate with HTTP endpoints with ease.
+
+There are a few ways to do requests - let's take a look at them
+
+### $http()
+
+We can use `$http` as a function, passing through a configuration object with it, as such:
+
+```js
+$http({
+	method: 'GET',
+	url: '/someURL'
+});
+
+$http({
+	method: 'POST',
+	data: {
+		username: 'Bill'
+	},
+	url: '/someOtherURL'
+});
+```
+
+This will return a promise with the data - we call the `.then` function that the function call returns - passing through a callback function that will get called when the request has finished.
+
+```js
+$http({
+	method: 'GET',
+	url: '/someURL'
+})
+	.then(function (data) {
+		console.log(data);
+	});
+```
+
+This will load the response from our URL, and allow us to consume the data somewhere after it's returned the response.
+
+### $http.get() and $http.post();
+
+$http also gives us helper functions instead of calling it as a function itself.
+
+They act similar to the example above, but as we already have the method in the name we can just pass through the URL we would like as a string.
+
+```js
+$http.get('/someURL');
+
+$http.post('someOtherURL', { username: 'Bill' });
+```
+
+### Using these in our services
+
+Now, any usage of `$http` should be done in a custom service that we create. This makes our controllers nice a thin, and allows us to call our API from anywhere in the application. If we didn't abstract our API calls out into a service, we might end up having two calls to the same endpoint somewhere in our application. If that endpoint then changed, we'd have to look around the application for any usage of the endpoint, instead of just changing it once in one file.
```

Comment:
Good explanation of the problem associated with using `$http` right in the controller. Can you add some context here about how this applies to `MVC` and/or `MVVM`?